examples/workspace-tags/main.tf

```diff
@@ -0,0 +1,170 @@
+terraform {
+  required_providers {
+    coder = {
+      source = "coder/coder"
+    }
+    docker = {
+      source = "kreuzwerker/docker"
+    }
+  }
+}
+
+locals {
+  username = data.coder_workspace.me.owner
+}
+
+data "coder_provisioner" "me" {
+}
+
+data "coder_workspace" "me" {
+}
+
+data "coder_workspace_tags" "custom_workspace_tags" {
+  tags = {
+    "zone"       = "developers"
+    "os"         = data.coder_parameter.os_selector.value
+    "project_id" = "PROJECT_${data.coder_parameter.project_name.value}"
+    "cache"      = data.coder_parameter.feature_cache_enabled.value == "true" ? "with-cache" : "no-cache"
+  }
+}
+
+data "coder_parameter" "os_selector" {
+  name         = "os_selector"
+  display_name = "OS runtime"
+  default      = "linux"
+
+  option {
+    name  = "Linux"
+    value = "linux"
+  }
+  option {
+    name  = "OSX"
+    value = "osx"
+  }
+  option {
+    name  = "Windows"
+    value = "windows"
+  }
+
+  mutable = false
+}
```

Comment:
Could you explain if `data.coder_parameter.os_selector` represents the os of the workspace being provisioned? This example confuses me in understanding the purpose and benefits of using `workspace_tags`. If yes, then the `code-server` and the `docker_container` can only run on a Linux host. This feels incomplete on how a template can provide a workspace with three different OS using the new functionality and provisions. 

As this will also be published as an official example on https://registry.coder.com/templates, we should try to make the tags as practical as possible.

Reply:
We can remove the example altogether, too.

Reply:
Renamed the `os_selector` to `runtime_selector`, so it might be easier to comprehend that is about selecting either a free zone or an isolated, secure, air-gapped environment. I hope it makes it clear.

BTW I can modify it to depend on "foobars" too. My intention was to indicate the syntax rather providing use cases.